core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -542,23 +619,76 @@ protected void doSetUserAttribute(String userName, String attributeName, String
     protected void doSetUserClaimValueWithID(String userID, String claimURI, String claimValue, String profileName)
             throws UserStoreException {
 
-        if (log.isDebugEnabled()) {
-            log.debug("doSetUserClaimValueWithID operation is not implemented in: " + this.getClass());
+        try {
+            String attributeName = getClaimAtrribute(claimURI, userID, null);
+            Map<String, String> userStoreAttributeValueMap = new HashMap<>();
+            userStoreAttributeValueMap.put(attributeName, claimValue);
+            processAttributesBeforeUpdateWithID(userID, userStoreAttributeValueMap, profileName);
+
+            for (Map.Entry<String, String> entry : userStoreAttributeValueMap.entrySet()) {
+                doSetUserAttributeWithID(userID, entry.getKey(), entry.getValue(), profileName);
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new UserStoreException(
+                    "Error occurred while getting the claim attribute for claimURI: " + claimURI + " of the user: "
+                            + userID, e);
         }
-        throw new NotImplementedException(
-                "doSetUserClaimValueWithID operation is not implemented in: " + this.getClass());
     }
 
     /**
-     * Set many user claim values
+     * Set many user claim values.
      *
-     * @param userName    The user name
-     * @param claims      Map of claim URIs against values
+     * @param userName    The user name.
+     * @param claims      Map of claim URIs against values.
      * @param profileName The profile name, can be null. If null the default profile is considered.
-     * @throws UserStoreException An unexpected exception has occurred
+     * @throws UserStoreException An unexpected exception has occurred.
      */
-    protected abstract void doSetUserClaimValues(String userName, Map<String, String> claims,
-                                                 String profileName) throws UserStoreException;
+    protected void doSetUserClaimValues(String userName, Map<String, String> claims,
+                                        String profileName) throws UserStoreException {
+
+        if (profileName == null) {
+            profileName = UserCoreConstants.DEFAULT_PROFILE;
+        }
+
+        // resolving claims to user store attributes
```

Comment:
shall we refactor this?  should start with a capital and end with "."? check-in other places as well